tasks/main.yml

```diff
@@ -1 +1,36 @@
 # role tasks
+
+- name: Find the current locale setting
+  slurp:
+    src: /etc/locale.conf
+  register: host_locale
+
+- name: update locale if needed
+  become: "{{ localectl_become }}"
+  become_user: "{{ localectl_become_user }}"
+  command: localectl set-locale LANG="{{ localectl_locale}}"
+  when: localectl_locale not in host_locale.content | b64decode
```

Comment:
Are these conditionals sufficiently narrow to avoid problems when likely comments are present in a config file? Do we want to somehow narrow this down?

Tying in with the comment above about how large the resulting file is, we might want to fetch the contents of the file currently by passing the results through a pipe to eliminate lines that are clearly wrong, commented, or otherwise might get in the way of these tests. A simple example might be something like this:

`grep <file> -v -e '^#' | sed -E -e 's/#.*$//' | grep <option name>`

The first grep will eliminate lines that begin with a comment character ("#").
The sed will strip out comments that live within a line.
The last section will return online the line (or lines) that include the name of the setting being searched for.

Running this pipe through the `shell` module in Ansible and registering the result will allow you to run this conditional and others with greater (not perfect, but greater) confidence that you're not slurping extra data across the network and also reduces the chance that a string like "us" that's anywhere in the comment section of a file might cause this role to misbehave.

Reply:
I have updated all the conditionals in the role to be much less risky.